packages/core/src/input/InputFileResolver.ts

```diff
@@ -157,13 +159,20 @@ export default class InputFileResolver {
   }
 
   private async readFiles(fileNames: string[]): Promise<File[]> {
-    const files = await Promise.all(fileNames.map((fileName) => this.readFile(fileName)));
-    return files.filter(notEmpty);
+    const files = from(fileNames)
+      .pipe(
+        mergeMap((fileName) => this.readFile(fileName), MAX_CONCURRENT_FILE_IO),
+        filter(notEmpty),
+        toArray()
+      )
+      .toPromise();
+    return files;
   }
 
   private async readFile(fileName: string): Promise<File | null> {
     try {
-      const content = await fs.readFile(fileName);
+      const p = fs.promises.readFile(fileName);
```

Comment:
is there a reason for making it 2 liner?

Reply:
yeah, was debugging it earlier (🦸‍♀️hacker by night). Removing it 👍

Reply:
Fixed ✅